ktor-shared/ktor-serialization/ktor-serialization-kotlinx/common/src/io/ktor/serialization/kotlinx/KotlinxSerializationConverter.kt

```diff
@@ -112,7 +112,15 @@ public class KotlinxSerializationConverter(
                 TextContent(content, contentType.withCharsetIfNeeded(charset))
             }
             is BinaryFormat -> {
-                val content = format.encodeToByteArray(serializer as KSerializer<Any?>, value)
+                val content = try {
```

Comment:
what is the reason for such check at all?

Reply:
as far as I see from current kx.serialization code: https://github.com/Kotlin/kotlinx.serialization/blob/dc950f5098162a3f8dd742d04b95f9a0405470e1/formats/protobuf/commonMain/src/kotlinx/serialization/protobuf/internal/ProtobufTaggedEncoder.kt#L36 there are at least 4 different messages for null related things

why not to just throw error when `value==null`?
Also, I don't think, that `NullIsNotSupportedSerializationException` should exists and be public - generic `SerializationException` for this should be enough, but it's up to you

Reply:
This was my idea to have a dedicated exception, because some formats support `null` objects and some are not. But it looks like there is no good way to implement it, so this idea can be dropped.

Reply:
ok, then I will rollback to the initial state